src/main/rx/cc2500_redpine.c

```diff
@@ -0,0 +1,194 @@
+    static timeUs_t packetTimerUs;
+
+    static timeUs_t totalTimerUs;
+ 
+    rx_spi_received_e ret = RX_SPI_RECEIVED_NONE;
+
+    switch (*protocolState) {
+    case STATE_STARTING:
+        listLength = 47;
+        initialiseData(0);
+        *protocolState = STATE_UPDATE;
+        nextChannel(1);
+        cc2500Strobe(CC2500_SRX);
+#ifdef USE_RX_FRSKY_SPI_PA_LNA
+        cc2500TxDisable();
+#endif // USE_RX_FRSKY_SPI_PA_LNA
+
+        break;
+    case STATE_UPDATE:
+        packetTimerUs = 0;
+        totalTimerUs = micros();
+        *protocolState = STATE_DATA;
+        if (rxSpiCheckBindRequested(false)) {
+            packetTimerUs = 0;
+            missingPackets = 0;
+            *protocolState = STATE_INIT;
+            break;
+        }
+
+        FALLTHROUGH;
+        // here FS code could be
+    case STATE_DATA:
+        if (cc2500getGdo()) {
+            uint8_t ccLen = cc2500ReadReg(CC2500_3B_RXBYTES | CC2500_READ_BURST) & 0x7F;
+            DEBUG_SET(DEBUG_RX_FRSKY_SPI, 3, ccLen);
+            if (ccLen > RX_SPI_MAX_PAYLOAD_SIZE) {
+                 cc2500Strobe(CC2500_SFRX);
+            } else if (ccLen) {
+                cc2500ReadFifo(packet, ccLen);
+                uint16_t lcrc= calculateCrc(&packet[0], CHANNEL_START+9);
+
+                if(((lcrc >> 8) == packet[CHANNEL_START+9]) && 
+                    ((lcrc&0x00FF) == packet[CHANNEL_START+10]) &&
+                    (packet[1] == rxFrSkySpiConfig()->bindTxId[0]) &&
+                    (packet[2] == rxFrSkySpiConfig()->bindTxId[1])) {
+                    
+                    looptime = packet[CHANNEL_START+7] * 100;
+
+                    DEBUG_SET(DEBUG_RX_FRSKY_SPI, 0, looptime);
+                    DEBUG_SET(DEBUG_RX_FRSKY_SPI, 1, packet[CHANNEL_START+8] * 10);
+                    packetTimerUs = micros();
+                    totalTimerUs = micros();                         
+
+                    missingPackets = 0;
+                    rxSpiLedOn();
+
+                    cc2500setRssiDbm(packet[ccLen - 2]);
+
+                    ret = RX_SPI_RECEIVED_DATA;
+                    nextChannel(1);
+                    cc2500Strobe(CC2500_SRX);
+                } 
+            }
+        }
+
+        if (cmpTimeUs(micros(), totalTimerUs) > 50 * looptime) {
+            //out of sync with packets - do a complete resysnc
+            rxSpiLedToggle();
+            setRssiDirect(0, RSSI_SOURCE_RX_PROTOCOL);
+            nextChannel(1);
+            cc2500Strobe(CC2500_SRX);
+            *protocolState = STATE_UPDATE;
+        } else if ((cmpTimeUs(micros(), packetTimerUs) > looptime) && packetTimerUs) {
+            //missed a packet
+            packetTimerUs = micros();
+            nextChannel(1);
+            cc2500Strobe(CC2500_SRX);
+            missingPackets++;
+            DEBUG_SET(DEBUG_RX_FRSKY_SPI, 2, missingPackets);
+#if defined(USE_RX_FRSKY_SPI_DIVERSITY)
+            if (missingPackets >= 2) {
+                if (missingPackets >= 2) {
```

Comment:
Does this variable need to be checked twice in a row?

Reply:
Nope!  Type-o, let me pull that out